ipykernel/tests/test_kernelspec.py

```diff
@@ -88,10 +90,41 @@ def test_write_kernel_spec_path():
     shutil.rmtree(path)
 
 
+@pytest.mark.skipif(sys.platform == 'win32',
+                    reason="does not run on windows")
+def test_write_kernel_spec_permissions(tmp_path):
```

Comment:
This function should now probably be the main focus of this PR. I believe the implementation from #593 runs on Windows (at least it does for me on Python 3.9.2), so no need to skip that platform.